packages/web-components/src/components/ic-top-navigation/ic-top-navigation.tsx

```diff
@@ -312,17 +323,30 @@ export class TopNavigation {
             <header role="banner">
               <div class="top-panel-container">
                 <div class="app-details-container">
-                  {hasTitle && (
-                    <a class="title-link" href={this.href}>
+                  {this.hasRouterAppTitle() ? (
+                    <div class="title-link" tabindex="0" role="link">
```

Comment:
To make things simpler, could we just allow the user to focus on the actual slotted link itself? I.e. remove the tabindex attributes, and then set the focus indicator styling by doing :focus-within on the title-link class. And then I don't think we would need the aria-label or role to be set as the screen reader will read out the content of the focussed link and the fact that it's a link since it's focussing on the `<a>`

Not 100% sure if it would work but maybe worth a try? Sorry if you've already given it a go :)